// undercore/index.js
const index = require('./hosts.js');

module.exports = {
	get : {
		hash : function getHashTags(inputText) {
			var regex = /(?:^|\s)(?:#)([가-힣a-zA-Z\d]+)/gm;
			var matches = [];
			var match;

			while ((match = regex.exec(inputText))) {
				matches.push(match[1]);
			}

			return matches;
		},
		filter : function(types, host){
			var len = types.length;
			var category = "";
			var channel = "";
			var sns = "";

			for(var i = 0; i < len; i++){
				var type = types[i].toLowerCase();

				if( type == "banner"  ||
					type == "show"    || 
					type == "music"   ||
					type == "goods"   ||
					type == "promotion"  ||
					type == "gallery" ||
					type == "diary"   ||
					type == "lifestyle"
				){
					category = type;
				}

				if( type == "mari" || 
					type == "lua"  ||
					type == "harang"
				){
					channel = type;
				}

				if( type == "twitter"   ||
					type == "instagram" ||
					type == "facebook"  ||
					type == "pinterest" ||
					type == "youtube"   ||
					type == "vimeo"
				){
					sns = type;
				}
			}

			category = category == "" ? "diary" : category;
			channel = channel == "" ? "team" : channel;

			return {
				category   : category,
				channel : channel,
				sns : sns
			}
		}
	}
}
